sample-extension-plugin/src/test/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunnerRestIT.java

```diff
@@ -220,4 +223,111 @@ public void testAcquiredLockPreventExecOfTasks() throws Exception {
+
+        // Run job and check for release = false
+        waitUntilLockIsAcquiredAndReleasedTransportCall(jobId, 20, LOCK_INFO_URI, navigationFunctionEntireCuster);
+
+        // Call the scheduled info API
+        // Checks lock is released
+        Response response = makeRequest(client(), "GET", LOCK_INFO_URI, Map.of(), null);
+        Map<String, Object> responseJson = JsonXContent.jsonXContent.createParser(
+            NamedXContentRegistry.EMPTY,
+            LoggingDeprecationHandler.INSTANCE,
+            response.getEntity().getContent()
+        ).map();
+
+        // Asserts that "released" is true
+        assertTrue(navigationFunctionEntireCuster.apply(responseJson));
+
+    }
+
+    public void testActiveLockTransportPathParameter() throws Exception {
+        String LOCK_INFO_URI = "/_plugins/_job_scheduler/api/locks";
+
+        String index = createTestIndex();
+        SampleJobParameter jobParameter = new SampleJobParameter();
+        jobParameter.setJobName("sample-job-lock-test-it");
+        jobParameter.setIndexToWatch(index);
+        jobParameter.setSchedule(new IntervalSchedule(Instant.now(), 5, ChronoUnit.SECONDS));
+        jobParameter.setLockDurationSeconds(10L);
+
+        String jobId = OpenSearchRestTestCase.randomAlphaOfLength(10);
+        createWatcherJob(jobId, jobParameter);
+
+        // Run job and check for release = false
+        waitUntilLockIsAcquiredAndReleasedTransportCall(jobId, 20, LOCK_INFO_URI, navigationFunctionEntireCuster);
+
+        // Call the scheduled info API
+        // Checks lock is released
+        LOCK_INFO_URI = "/_plugins/_job_scheduler/api/locks/" + ".scheduler_sample_extension" + "-" + jobId;
+
+        Response response = makeRequest(client(), "GET", LOCK_INFO_URI, Map.of(), null);
+        Map<String, Object> responseJson = JsonXContent.jsonXContent.createParser(
+            NamedXContentRegistry.EMPTY,
+            LoggingDeprecationHandler.INSTANCE,
+            response.getEntity().getContent()
+        ).map();
+
+        // Asserts that "released" is true
+        assertTrue(navigationFunctionEntireCuster.apply(responseJson));
+    }
+
+    protected void waitUntilLockIsAcquiredAndReleasedTransportCall(
+        String jobId,
+        int maxTimeInSec,
+        String SCHEDULER_INFO_URI,
+        Function<Map<String, Object>, Boolean> navigationFunction
+    ) throws IOException, InterruptedException {
+        AtomicLong prevLockAcquiredTime = new AtomicLong(0L);
+        AtomicReference<LockModel> lock = new AtomicReference<>();
+
+        try {
+            lock.set(getLockByJobId(jobId));
+            if (lock.get() != null && prevLockAcquiredTime.get() == 0L && lock.get().isReleased()) {
+                prevLockAcquiredTime.set(lock.get().getLockTime().toEpochMilli());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Ensure the job is running then calls API
+        Thread.sleep(7000);
```

Comment:
Can this be refactored to avoid using sleep? i.e. using Awaitility to wait until there's an entry in the lock index and `released` is set to false?